### ISSUE TITLE
When searching for versions, if no groupId exists, assume the default

### ISSUE THREAD

**Issue by anthonyvdotbe** [microsoft/vscode-maven#594]
**Describe the bug**
If I have:
```xml
<plugin>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.8.1</version>
</plugin>
```
and list the possible versions (with `Ctrl`+`space`), it lists version 3.9.0.
Apparently, it searches just by artifactId and lists an unwanted artifact (see https://search.maven.org/search?q=a:maven-compiler-plugin).
However, for plugins, if the `groupId` isn't specified, the default is `org.apache.maven.plugins`. So in this case, it should search by default groupId and artifactId.

**To Reproduce**
1. Create a POM with the above snippet under `plugins`
2. Trigger autocompletion for the version
3. Notice that `3.9.0` is listed

**Expected behavior**
For plugins, if the `groupId` isn't specified, it is `org.apache.maven.plugins`. So in this case, `3.9.0` shouldn't be listed.

**Environments (please complete the following information as much as possible):**
 - OS: Windows 10
 - VS Code version: 1.54.3
 - Extension version: 0.28.0

**Comment by Eskibear** [comment]
It was designed as a feature like fuzzy searching. E.g. If I don't remember clearly about group id, by searching keywords of artifact id, I can find the wanted entry. Corresponding group id is shown in details of each completion item, and in the end after you select it, group id is also auto-completed. 

Do you think it still bother you or confuse you, and insists to use `org.apache.maven.plugins` as implied group id in this case?

**Comment by Eskibear** [comment]
As for completion for `version` nodes, I agree with what you expect above. It is strange that the dependency/plugin changes to another after I select a version...

**Comment by anthonyvdotbe** [comment]
It seems you've already fixed my issue, but just for clarification: say I have a POM which defines the `maven-compiler-plugin`. Now I want to quickly check if there are any plugin updates, so I perform auto-completion:

![image](https://user-images.githubusercontent.com/15714253/113853036-7e539a80-979d-11eb-9cbc-627dd359b5bf.png)

Seeing a version of `3.9.0`, I assume there's an update and set that as the version. But then next time I build the project, it fails because there is no version 3.9.0 of `org.apache.maven.plugins:maven-compiler-plugin`.

So I'd expect the auto-completion to not show version 3.9.0 in the list in this case, which is exactly what #609 does, right? If so, other than that everything works great, so this issue can be closed then.

**Comment by Eskibear** [comment]
It makes sense. And PR #609 fixed the issue, as shown in its description. I'll close this issue after we ship it next week.

**Comment by Eskibear** [comment]
shipped in v0.29.0
